### PR TITLE
Clean up loose ends with environment files 

### PR DESCRIPTION
**Pull request by carolinemodic**
## Overview
https://github.com/votingworks/vxsuite/issues/2796
This PR does a couple of small clean ups to improve how we're handling environment variables and files

- Fixes the pnpm call to run the generation script to create a .env.local file
- Moves the logic for the VxAdmin converter environment variable to the shared utility. Since this is the only string variable we currently have and we want it strongly typed I didn't fully generalize this yet. But by moving it here we are able to include that variable in the script that generates .env.local files.  In the process of doing this I renamed some of the boolean variable functions/types, feel free to suggest better names. 
- Updates the default .env file at the root level to be all of the default values when you run the env file generation script (in the future we should figure out how to automate this being updated or have a test to enforce its updated cc @amcmanus ) 
- Removes the now completely redundant and unnecessary subrepo default .env files. 

## Testing Plan 
Verified that I can still set the convertor with .env or .env.local files properly. Ran the generation script with valid and invalid inputs. 

## Checklist
- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [ ] I have added JSDoc comments to any newly introduced exports

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -116,6 +116,7 @@ jobs:
       - run:
           name: Build
           command: |
+            cp .env frontends/election-manager/.env
```
Comment (carolinemodic) [inline .circleci/config.yml:119]: I feel like there must be a better solution to this problem but for some reason circleCI doesn't seem to be looking at the root-level .env file, which only matters for getting the ms-sems converter value in VxAdmin, this fixed it but I don't know why I need to do this.

File: .env [hunk 2]
```diff
@@ -2,3 +2,7 @@ REACT_APP_VX_ENABLE_WRITE_IN_ADJUDICATION=TRUE
 REACT_APP_VX_ENABLE_ALL_ZERO_SMARTCARD_PIN_GENERATION=TRUE
 REACT_APP_VX_DISABLE_CARD_READER_CHECK=FALSE
 REACT_APP_VX_ENABLE_LIVECHECK=TRUE
+REACT_APP_VX_DISALLOW_CASTING_OVERVOTES=FALSE
```
Comment (umbernhard) [inline .env:5]: this is a huge nitpick but would it be possible to make this `ALLOW_CASTING_OVERVOTES` so it wouldn't be a double negative?

File: .circleci/config.yml (hunk 1 above)
Comment (umbernhard) [reply .circleci/config.yml:119]: FWIW it looks like this file did exist but is deleted by this PR.

**Review by umbernhard — APPROVED** [review]
Looks good to me, a few nitpicks here and there.

File: .circleci/config.yml (hunk 1 above)
Comment (carolinemodic) [reply .circleci/config.yml:119]: yeah but it shouldn't be necessary

File: .env (hunk 2 above)
Comment (carolinemodic) [reply .env:5]: out of scope of this pr

**Pull request merged**